pkg/event/target/mqtt.go

```diff
@@ -134,19 +164,50 @@ func NewMQTTTarget(id string, args MQTTArgs) (*MQTTTarget, error) {
 		SetTLSConfig(&tls.Config{RootCAs: args.RootCAs}).
 		AddBroker(args.Broker.String())
 
+	var store Store
+
 	if args.QueueDir != "" {
-		options = options.SetStore(mqtt.NewFileStore(args.QueueDir))
+		store = NewQueueStore(args.QueueDir, 10000)
```

Comment:
perhaps you need to take args.QueueLimit? here

Reply:
Yes that is a work in progress. Will update soon